### ISSUE TITLE
List conversion problem

### ISSUE THREAD

**Issue by flexibeast** [makeworld-the-better-one/md2gemini#22]
Thanks for md2gemini - it's looking great!

Myself and a fellow member of the Void Linux docs team would like to make [the Void Handbook](https://docs.voidlinux.org/) available in geminispace, and md2gemini seems like it could it could significantly facilitate this. :-)

However, running [about/index.md](https://raw.githubusercontent.com/void-linux/void-docs/master/src/about/index.md) through md2gemini takes a source list that looks like:

    - The [XBPS](https://github.com/void-linux/xbps) package manager, which is
       extremely fast, developed in-house, and performs checks when installing
       updates to ensure that libraries are not changed to incompatible versions
       which can break dependencies.
    - The [musl libc](https://musl.libc.org/), which focuses on standards compliance
       and correctness, has first class support. This allows us to build certain
       components for musl systems statically, which would not be practical on glibc
       systems.
    - The [LibreSSL](https://www.libressl.org/) fork is used instead of the mainline
       OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is
       dedicated to the security, quality, and maintainability of this critical
       library.

and produces:

    * The XBPS[6] package manager, which is
    * extremely fast, developed in-house, and performs checks when installing
    * updates to ensure that libraries are not changed to incompatible versions
    * which can break dependencies.
    * The musl libc[7], which focuses on standards compliance
    * and correctness, has first class support. This allows us to build certain
    * components for musl systems statically, which would not be practical on glibc
    * systems.
    * The LibreSSL[8] fork is used instead of the mainline
    * OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is
    * dedicated to the security, quality, and maintainability of this critical
    * library.

The Markdown used for the Handbook is [Versioned Markdown](https://github.com/bobertlo/vmd).

cc: @ericonr

**Comment by makeworld-the-better-one** [comment]
Thanks for finding this, definitely unexpected behaviour. I will look into fixing it. However I am having trouble reproducing this. Can you show me the exact commandline or python options you used?

Also note that the expected output would be to have the entire content of the bullet point on one line. This is not what is happening when I test this, which is still a bug, but seems to be a different one than you are experiencing.

This is my output with the links mode `at-end`.
```
* The XBPS[1] package manager, which is
 extremely fast, developed in-house, and performs checks when installing
 updates to ensure that libraries are not changed to incompatible versions
 which can break dependencies.
* The musl libc[2], which focuses on standards compliance
 and correctness, has first class support. This allows us to build certain
 components for musl systems statically, which would not be practical on glibc
 systems.
* The LibreSSL[3] fork is used instead of the mainline
 OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is
 dedicated to the security, quality, and maintainability of this critical
 library.


=> https://github.com/void-linux/xbps 1: https://github.com/void-linux/xbps
=> https://musl.libc.org/ 2: https://musl.libc.org/
=> https://www.libressl.org/ 3: https://www.libressl.org/
```

The correct output would be:
```
* The XBPS[1] package manager, which is extremely fast, developed in-house, and performs checks when installing updates to ensure that libraries are not changed to incompatible versions which can break dependencies.
* The musl libc[2], which focuses on standards compliance and correctness, has first class support. This allows us to build certain components for musl systems statically, which would not be practical on glibc systems.
* The LibreSSL[3] fork is used instead of the mainline OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is dedicated to the security, quality, and maintainability of this critical library.


=> https://github.com/void-linux/xbps 1: https://github.com/void-linux/xbps
=> https://musl.libc.org/ 2: https://musl.libc.org/
=> https://www.libressl.org/ 3: https://www.libressl.org/
```

**Comment by makeworld-the-better-one** [comment]
Nevermind, I can reproduce when I use the entire document:
```shell
curl -sSL https://raw.githubusercontent.com/void-linux/void-docs/master/src/about/index.md | python3 -m md2gemini -l at-end
```

I will look into what is causing, thanks for the report.

**Comment by flexibeast** [comment]
Oh, my apologies - poor form of me to forget to include the command i ran. :-( Sorry to waste your time! The specific command was `md2gemini --links paragraph index.md` (as `paragraph` mode is the one we'd like to use).

**Comment by ericonr** [comment]
Hmm I believe this might be an issue with `mistune` instead...

**Comment by makeworld-the-better-one** [comment]
```
~/Downloads
❯ curl -sSL https://raw.githubusercontent.com/void-linux/void-docs/master/src/about/index.md > void.md

~/Downloads
❯ python3
Python 3.8.5 (default, Sep  5 2020, 10:50:12) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mistune
>>> mistune.__version__
'2.0.0a4'
>>> f = open("void.md", "r")
>>> mistune.html(f.read())
'<h1>About</h1>\n<p>Welcome to the Void Handbook! Please be sure to read the &quot;<a href="./about-this-handbook.md">About This\nHandbook</a>&quot; section to learn how to use this\ndocumentation effectively. A local copy of this handbook, in several formats,\ncan be <a href="../xbps/index.md">installed</a> via the <code>void-docs</code> package and accessed\nwith the <a href="https://man.voidlinux.org/void-docs.1">void-docs(1)</a> utility.</p>\n<p>Void is an independent, <a href="https://en.wikipedia.org/wiki/Rolling_release">rolling\nrelease</a> Linux distribution,\ndeveloped from scratch rather than as a fork, with a focus on stability over\n<a href="https://en.wikipedia.org/wiki/Bleeding_edge_technology">bleeding-edge</a>. In\naddition, there are several features that make Void unique:</p>\n<ul>\n<li>The <a href="https://github.com/void-linux/xbps">XBPS</a> package manager, which is\n extremely fast, developed in-house, and performs checks when installing\n updates to ensure that libraries are not changed to incompatible versions\n which can break dependencies.</li>\n<li>The <a href="https://musl.libc.org/">musl libc</a>, which focuses on standards compliance\n and correctness, has first class support. This allows us to build certain\n components for musl systems statically, which would not be practical on glibc\n systems.</li>\n<li>The <a href="https://www.libressl.org/">LibreSSL</a> fork is used instead of the mainline\n OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is\n dedicated to the security, quality, and maintainability of this critical\n library.</li>\n<li><a href="../config/services/index.md">runit</a> is used for\n <a href="https://man.voidlinux.org/init.8">init(8)</a> and service supervision. This\n allows Void to support musl as a second libc choice, which would not be\n possible with <a href="https://www.freedesktop.org/wiki/Software/systemd/">systemd</a>.\n A side effect of this decision is a core system with clean and efficient\n operation, and a small code base.</li>\n</ul>\n<p>Void is developed in the spare time of a handful of developers, and is generally\nconsidered stable enough for daily use. We do this for fun and hope that our\nwork will be useful to others.</p>\n<p>The name &quot;Void&quot; comes from the C literal <code>void</code>. It was chosen rather randomly,\nand has no deeper meaning.</p>\n'
>>> f.close()
```

Formatted HTML:
```html
<h1>About</h1>
<p>Welcome to the Void Handbook! Please be sure to read the &quot;<a href="./about-this-handbook.md">About This
Handbook</a>&quot; section to learn how to use this
documentation effectively. A local copy of this handbook, in several formats,
can be <a href="../xbps/index.md">installed</a> via the <code>void-docs</code> package and accessed
with the <a href="https://man.voidlinux.org/void-docs.1">void-docs(1)</a> utility.</p>
<p>Void is an independent, <a href="https://en.wikipedia.org/wiki/Rolling_release">rolling
release</a> Linux distribution,
developed from scratch rather than as a fork, with a focus on stability over
<a href="https://en.wikipedia.org/wiki/Bleeding_edge_technology">bleeding-edge</a>. In
addition, there are several features that make Void unique:</p>
<ul>
<li>The <a href="https://github.com/void-linux/xbps">XBPS</a> package manager, which is
 extremely fast, developed in-house, and performs checks when installing
 updates to ensure that libraries are not changed to incompatible versions
 which can break dependencies.</li>
<li>The <a href="https://musl.libc.org/">musl libc</a>, which focuses on standards compliance
 and correctness, has first class support. This allows us to build certain
 components for musl systems statically, which would not be practical on glibc
 systems.</li>
<li>The <a href="https://www.libressl.org/">LibreSSL</a> fork is used instead of the mainline
 OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is
 dedicated to the security, quality, and maintainability of this critical
 library.</li>
<li><a href="../config/services/index.md">runit</a> is used for
 <a href="https://man.voidlinux.org/init.8">init(8)</a> and service supervision. This
 allows Void to support musl as a second libc choice, which would not be
 possible with <a href="https://www.freedesktop.org/wiki/Software/systemd/">systemd</a>.
 A side effect of this decision is a core system with clean and efficient
 operation, and a small code base.</li>
</ul>
<p>Void is developed in the spare time of a handful of developers, and is generally
considered stable enough for daily use. We do this for fun and hope that our
work will be useful to others.</p>
<p>The name &quot;Void&quot; comes from the C literal <code>void</code>. It was chosen rather randomly,
and has no deeper meaning.</p>
```

**Rendered version (bullet points only):**

<ul>
<li>The <a href="https://github.com/void-linux/xbps">XBPS</a> package manager, which is
 extremely fast, developed in-house, and performs checks when installing
 updates to ensure that libraries are not changed to incompatible versions
 which can break dependencies.</li>
<li>The <a href="https://musl.libc.org/">musl libc</a>, which focuses on standards compliance
 and correctness, has first class support. This allows us to build certain
 components for musl systems statically, which would not be practical on glibc
 systems.</li>
<li>The <a href="https://www.libressl.org/">LibreSSL</a> fork is used instead of the mainline
 OpenSSL library. Developed as part of the OpenBSD project, LibreSSL is
 dedicated to the security, quality, and maintainability of this critical
 library.</li>
<li><a href="../config/services/index.md">runit</a> is used for
 <a href="https://man.voidlinux.org/init.8">init(8)</a> and service supervision. This
 allows Void to support musl as a second libc choice, which would not be
 possible with <a href="https://www.freedesktop.org/wiki/Software/systemd/">systemd</a>.
 A side effect of this decision is a core system with clean and efficient
 operation, and a small code base.</li>
</ul>

@ericonr It appears that mistune is parsing this fine, and it's a bug in this repo's code. Would be interested if you have a code example that shows otherwise though, it would be very helpful for solving this bug.

**Comment by ericonr** [comment]
Probably not, it's the renderer's `list_item` that needs to be a bit more complex. I'm attempting a fix.

**Comment by ericonr** [comment]
Had missed your post! And yes, you're right, it's the renderer here. I believe I have an idea for a fix.

**Comment by ericonr** [comment]
@makeworld-the-better-one since we're here, is this line correct: https://github.com/makeworld-the-better-one/md2gemini/blob/master/md2gemini/renderers.py#L193 ? The comment describes something different from what the code is doing.

**Comment by makeworld-the-better-one** [comment]
The comment is actually a testing line that's been commented out. When I want to see what's being detected as a newline, I uncomment it and comment the other one. 

An empty string is returned in that case because Markdown allows you to have new lines that are part of the same paragraph, while gemtext requires that a paragraph all be on the same line.

**Comment by makeworld-the-better-one** [comment]
This is now available in a release on PyPI, v1.7.0. Happy to be able to help with the Void Linux project, feels cool.
